src/zope/app/apidoc/tests.py

```diff
@@ -238,14 +240,21 @@ def test_keyword_only_arguments(self):
                 getFunctionSignature(
                     TestUtilities.test_keyword_only_arguments))
         else:
-            self.assertEqual(simple_sig, "(mode='r', bufsize=-1)")
+            if six.PY3:
+                self.assertEqual(
+                    simple_sig, "(self, mode='r', buffering=None)")
+                self.assertEqual(
+                    '(self)', getFunctionSignature(
+                        TestUtilities.test_keyword_only_arguments))
+            else:
+                self.assertEqual(simple_sig, "(mode='r', bufsize=-1)")
+                self.assertEqual(
+                    '()', getFunctionSignature(
+                        TestUtilities.test_keyword_only_arguments))
```

Comment:
AFAICT previously we had _simpleGetFunctionSignature() broken on Python 3, when called on socket.makefile(), and thus the Python 3 specific test assertions were in the `except:` branch leaving this `else:` branch aiming at Python 2.

Now you fixed `_simpleGetFunctionSignature(socket.makefile` to work on Python 3, which suggests that the `try:` block and the entire `except:` branch can be removed.  Am I right?

Reply:
Actually, things are even simpler than I thought: `_simpleGetFunctionSignature()` is only used when `inspect.signature` does not exist, i.e. on Python <= 3.2.  Why not `@unittest.skipIf(six.PY3)` the entire test?

Reply:
Thank you for catching this issue. I adapted the code in https://github.com/zopefoundation/zope.app.apidoc/pull/38/commits/61de1109be0e8a45ea6dbc2a64f1d4bf87e47940 but Python 2 is no longer supported by tox 4, so I'll drop Python 2 support.